src/kani-compiler/src/codegen_cprover_gotoc/context/vtable_ctx.rs

```diff
@@ -117,9 +118,10 @@ impl<'tcx> GotocCtx<'tcx> {
             format!("restricted_call_label_{}", self.vtable_ctx.get_call_site_global_idx()).into();
 
         // We only have the Gotoc type, we need to normalize to match the MIR type.
-        assert!(trait_ty.is_struct_tag());
-        let mir_name =
-            self.normalized_trait_name(*self.type_map.get(&trait_ty.tag().unwrap()).unwrap());
+        // Retrieve the MIR for `&dyn T` and normalize the name.
+        assert!(trait_ref.is_struct_tag());
+        let mir_type = self.type_map.get(&trait_ref.tag().unwrap()).unwrap();
+        let mir_name = self.normalized_trait_name(pointee_type(*mir_type).unwrap());
```

Comment:
Also, I know we called them this already, but `mir_type` and `mir_name` aren't the most descriptive variable names

Reply:
I'm all in to improve our variable names! Let me change that.